src/Tester/StorageTestConstants.cs

```diff
@@ -0,0 +1,144 @@
+
+        private const string DEFAULT_STORAGE_DATA_CONNECTION_STRING = "UseDevelopmentStorage=true";
+
+        public static void Init()
+        {
+            if (DataConnectionString != null)
+            {
+                return; // already initialized
+            }
+            if (File.Exists(ORLEANS_TEST_STORAGE_KEY_FILE_NAME))
+            {
+                using (TextReader input = File.OpenText(ORLEANS_TEST_STORAGE_KEY_FILE_NAME))
+                {
+                    string line = input.ReadToEnd();
+                    line = line.Trim();
+                    if (!String.IsNullOrEmpty(line))
+                    {
+                        Console.Out.WriteLine("Found the {0} file and using the Storage Key from there.", ORLEANS_TEST_STORAGE_KEY_FILE_NAME);
+                        Trace.WriteLine("Found the {0} file and using the Storage Key from there.", ORLEANS_TEST_STORAGE_KEY_FILE_NAME);
+                        DataConnectionString = line;
+                    }
+                }
+            }
+
+            if (DataConnectionString != null) return;
+
+            // If did not find the file, just use the DevelopmentStorage
+            Console.Out.WriteLine("Did not find the {0} file or it was empty. Using Default Storage Data Connection String instead.", ORLEANS_TEST_STORAGE_KEY_FILE_NAME);
+            Trace.WriteLine("Did not find the {0} file or it was empty. Using Default Storage Data Connection String instead.", ORLEANS_TEST_STORAGE_KEY_FILE_NAME);
+            DataConnectionString = DEFAULT_STORAGE_DATA_CONNECTION_STRING;
+        }
+
+        public static FileInfo GetDbFileLocation(DirectoryInfo dbDir, string dbFileName)
+        {
+            if (!dbDir.Exists) throw new FileNotFoundException("DB directory " + dbDir.FullName + " does not exist.");
+
+            string dbDirPath = dbDir.FullName;
+            var dbFile = new FileInfo(Path.Combine(dbDirPath, "TestDb.mdf"));
+            Console.WriteLine("DB file location = {0}", dbFile.FullName);
+
+            // Make sure we can write to local copy of the DB file.
+            MakeDbFileWriteable(dbFile);
+
+            return dbFile;
+        }
+
+        public static string GetSqlConnectionString(TestContext context)
+        {
+            string dbFileName = @"TestDb.mdf";
+            string dbDirPath = context.DeploymentDirectory;
+            return GetSqlConnectionString(new DirectoryInfo(dbDirPath), dbFileName);
+        }
+        public static string GetSqlConnectionString(DirectoryInfo dbDir, string dbFileName)
+        {
+            var dbFile = GetDbFileLocation(dbDir, dbFileName);
+
+            Console.WriteLine("DB directory = {0}", dbDir.FullName);
+            Console.WriteLine("DB file = {0}", dbFile.FullName);
+
+            string connectionString = string.Format(
+                @"Data Source=(LocalDB)\v11.0;"
+                + @"AttachDbFilename={0};"
```

Comment:
Ah, table creation, I'm a bit hand-waving here, but it'd be something like:


        public void Xyz()
        {            
            string sqlDatabaseCreation = File.ReadAllText("CreateTables.sql");
            using(var conn = new SqlConnection(@"Data Source=(LocalDb)\ProjectsV12;Initial Catalog=OrleansTest;Integrated Security=True"))
            {
                conn.Open();
                using(var cmd = new SqlCommand())
                { 
                    cmd.Connection = conn;
                    cmd.CommandText =  string.Format(sqlDatabaseCreation);
                    cmd.ExecuteNonQuery();
                }                
            }
        }

This could be done, say, once in the beginning. If there's a need, it's of course possible to take backups (wrapped to a singleton that owns the backup) of the database and restore them. I don't see a need for that currently, but possible nevertheless